fdbclient/ManagementAPI.actor.cpp

```diff
@@ -465,6 +467,160 @@ bool isCompleteConfiguration(std::map<std::string, std::string> const& options)
 	       options.count(p + "storage_engine") == 1;
 }
 
+bool validTenantAndEncryptionAtRestMode(Optional<DatabaseConfiguration> oldConfiguration,
+                                        std::map<std::string, std::string> newConfig,
+                                        bool creating) {
+	if (creating) {
+		if (newConfig.count(encryptionAtRestModeConfKey.toString()) != 0) {
+			EncryptionAtRestMode encryptMode = EncryptionAtRestMode::fromValueRef(
+			    ValueRef(newConfig.find(encryptionAtRestModeConfKey.toString())->second));
+			if (encryptMode.mode == EncryptionAtRestMode::DISABLED ||
+			    encryptMode.mode == EncryptionAtRestMode::CLUSTER_AWARE) {
+				// In these cases the tenantMode does not matter
+				return true;
+			}
+
+			// check if the tenant mode is being set during configure new (otherwise assume tenants are disabled)
+			TenantMode tenantMode = TenantMode::DISABLED;
+			if (newConfig.count(tenantModeConfKey.toString()) != 0) {
+				tenantMode = TenantMode::fromValue(ValueRef(newConfig.find(tenantModeConfKey.toString())->second));
+			}
+			TraceEvent(SevDebug, "EncryptAndTenantModes")
+			    .detail("EncryptMode", encryptMode.toString())
+			    .detail("TenantMode", tenantMode.toString());
+			if (tenantMode != TenantMode::REQUIRED) {
+				// For domain aware encryption only the required tenant mode is currently supported
+				TraceEvent(SevWarnAlways, "InvalidEncryptAndTenantConfiguration")
+				    .detail("EncryptMode", encryptMode.toString())
+				    .detail("TenantMode", tenantMode.toString());
+				return false;
+			}
+			return true;
+		}
+	} else {
```

Comment:
should we also add:

`newConfig.count(encryptionAtRestModeConfKey.toString()) != 0 {return false}`

We don't want EaR mode to get updated once its set (at the creation time)?

Reply:
You already do this here: https://github.com/apple/foundationdb/blob/main/fdbclient/include/fdbclient/GenericManagementAPI.actor.h#L279